subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/moduleconverter/DefaultRootComponentMetadataBuilder.java

```diff
@@ -134,4 +144,44 @@ DefaultLocalComponentMetadata tryCached(ComponentIdentifier id) {
             return null;
         }
     }
+
+
+    public static class Factory {
```

Comment:
I'd think a top-level `DefaultRootComponentMetadataBuilderFactory` class might be a little clearer than a class just named `Factory` which could potentially be confused with other imported factories by client classes, but I don't feel strongly about.

Reply:
This pattern of putting the factory inside of the same class is pretty common. From both a readability and a "jump to declaration" perspective it makes navigating the code much easier.